docs/docs/configuration/secrets-manager.md

```diff
@@ -0,0 +1,91 @@
+---
+sidebar_position: 4
+---
+
+# How to use AWS Secret Manager, GCP Secret Manager, & Azure Key Vault
+
+Secrets from **AWS Secret Manager**, **GCP Secret Manager**, & **Azure Key Vault** can be used in any **Secure** field which looks like the following:
+
+![Example banner](/img/secure-field.png)
+
+* [AWS Secrets Manager](#aws-secrets-manager-setup)
+* [GCP Secret Manager](#gcp-secret-manager-setup)
+* [Azure Key Vault](#azure-key-vault-setup)
+* [Install all Secret Managers](#install-all-secret-managers)
+
+
+## AWS Secrets Manager Setup
+To use AWS Secrets Manager, you will need to install the `aws-secrets` dependency group. For example:
+```bash
+poetry install --extras "postgres redshift mysql trino athena snowflake" --with aws-secrets
```

Comment:
Makes me think that a group for each database driver could be nice in terms of developer experience 🙂 That way, we could event easily pin versions to prevent problems.

Reply:
Would you mind elaborating on this? Originally, I had hoped to add the secrets dependencies as Poetry `extras` but unfortunately you can't make extras optional so used dependency groups instead.

How would database driver dependency groups improve the developer experience?

Reply:
What I meant is that you could define groups for database drivers, the same way you did for secrets. For example:

```toml
[tool.poetry.group.postgres]
optional = true
[tool.poetry.group.postgres.dependencies]
postgres = "^X.X.X"
```

Then, the developer could install the PostgreSQL driver like this:

```sh
poetry install --with postgres --with aws-secrets
```

Reply:
So the improvement would be only using `--with` instead of `--extras` and `--with`?

Reply:
I updated the pyproject.toml to use dependency groups instead of extras.

In the docs, I intentionally added `---with secrets-provider` so that it is obvious what is being added.